suite-native/trading-residence/src/components/ResidenceCheckAwareAnimatedBox.tsx

```diff
@@ -0,0 +1,27 @@
+import { StretchInY, StretchOutY } from 'react-native-reanimated';
+import { useSelector } from 'react-redux';
+
+import { AnimatedBox, BoxProps } from '@suite-native/atoms';
+import { selectIsTradingResidenceCheckEnabled } from '@suite-native/trading-state';
+import { prepareNativeStyle, useNativeStyles } from '@trezor/styles';
+
+export type ResidenceCheckAwareAnimatedBoxProps = Omit<BoxProps, 'style'>;
+
+const pickerStyle = prepareNativeStyle(({ borders, colors }) => ({
```

Comment:
nit: you can do the this in prepareNativeStyle. Imo it will make it more readable.

```suggestion
const pickerStyle = prepareNativeStyle<hasBorder: boolean;>(({ borders, colors }, { hasBorder }) => ({
extend: [
        {
            condition: hasBorder,
            style: {
                borderTopWidth: borders.widths.small,
    		 	borderTopColor: colors.backgroundSurfaceElevation0,
            },
        },
    ],
```

Reply:
I opted in for one liner instead of extend, but will follow the team vote 989fa6ff54d0718ad6ee18186ae8643d6afa7562